cli/internal/specs/v0/spec.go

```diff
@@ -83,7 +83,22 @@ func (s *Spec) UnmarshalJSON(data []byte) error {
 	dec = json.NewDecoder(bytes.NewReader(b))
 	dec.UseNumber()
 	dec.DisallowUnknownFields()
-	return dec.Decode(s.Spec)
+	if err = dec.Decode(s.Spec); err != nil {
+		return err
+	}
+
+	var r struct {
+		Registry string `json:"registry"`
+	}
+	if err := json.Unmarshal(b, &r); err == nil && r.Registry == "" {
```

Comment:
The gist of it: During unmarshal, if `registry` (as a string) was empty, we mark it as such.

Reply:
Can't we make the JSON registry property a pointer then check for nullability instead?

Reply:
We can but it's everywhere, lots of places will have `to.Ptr` or `RegistryPtr` and dereferencing, which is not nice. I would rather make it a string if we're not doing this.

Reply:
@erezrokah https://github.com/cloudquery/cloudquery/pull/15409/commits/f99f20e56101e8c33eb91df4090fa06f9bcd0a58 uses `string` for `Registry`, no funny business.